### PR TITLE
Add foreman::providers to install provider dependencies

### PR DESCRIPTION
**Pull request by domcleal**
This will replace the package resource in foreman_proxy and help when
using the providers here outside of foreman_proxy.

### CODE REVIEW

File: manifests/providers.pp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#
+# $apipie_bindings_package::  Name of apipie-bindings package
+#
+# $foreman_api::              Install foreman_api dependency (deprecated)
+#                             type:boolean
+#
+# $foreman_api_package::      Name of foreman_api package
+#
+class foreman::providers(
+  $apipie_bindings         = $::foreman::providers::params::apipie_bindings,
```
Comment (ekohl) [inline manifests/providers.pp]: Validate the parameters using ```validate_bool``` and ```validate_string```?

Comment (domcleal) [inline manifests/providers.pp]: Thanks @ekohl, updated with the validation functions.

**Comment by mmoll** [comment]
Maybe the old foreman_api provider can get removed and the modules major bumped? I'm also OK with it being kept for now...
:+1: otherwise

**Comment by domcleal** [comment]
I'm not sure it's worth doing a major bump over... so I've added some docs to the README about the types/providers (long overdue) and noted there and in the provider header that it's deprecated so we can easily get rid of it later.

**Comment by mmoll** [comment]
merged, thanks @domcleal!

**Pull request merged**